File: myblog-server/src/main/java/com/site/blog/service/MailService.java
package com.site.blog.service;

import com.baomidou.mybatisplus.extension.service.IService;
import com.site.blog.model.entity.EmailConfig;

import java.util.Map;

/**
 * @Author: Yangzhengqian
 * @Description:
 * @Date:Created time 2020/11/12 上午10:46
 * @Modified By:
 */

public interface MailService  extends IService<EmailConfig> {
EmailConfig getDefaultMail();
    void sendTemplateEmail(String to, String subject, Map<String, Object> content);

    void sendSimpleMail(String to, String subject, String content);

    void sendHtmlMail(String to, String subject, String content);

    void sendAttachmentsMail(String to, String subject, String content, String filePath);

    void sendInlineResourceMail(String to, String subject, String content, String rscPath, String rscId);
    void sendFindPassEmail(String toEmail, EmailConfig emailConfig);


}

File: myblog-jpa/src/main/kotlin/com/site/blog/service/MailService.kt
package com.site.blog.service


import com.site.blog.model.entity.EmailConfig

/**
 * @Author: Yangzhengqian
 * @Description:
 * @Date:Created time 2020/11/12 上午10:46
 * @Modified By:
 */
interface MailService  {
    val defaultMail: EmailConfig?
    fun sendTemplateEmail(to: String?, subject: String?, content: Map<String?, Any?>?)
    fun sendSimpleMail(to: String?, subject: String?, content: String?)
    fun sendHtmlMail(to: String?, subject: String?, content: String?)
    fun sendAttachmentsMail(to: String?, subject: String?, content: String?, filePath: String?)
    fun sendInlineResourceMail(to: String?, subject: String?, content: String?, rscPath: String?, rscId: String?)
    fun sendFindPassEmail(toEmail: String?, emailConfig: EmailConfig?)
}